processor/src/host/advice/mod.rs

```diff
@@ -52,9 +52,9 @@ pub trait AdviceProvider: Sized {
     // --------------------------------------------------------------------------------------------
 
     /// Handles the specified advice injector request.
-    fn set_advice<S: ProcessState>(
+    fn set_advice(
         &mut self,
-        process: &S,
+        process: ProcessState,
         advice_injector: &AdviceInjector,
     ) -> Result<HostResponse, ExecutionError> {
```

Comment:
Here, and in other similar places, we should probably pass `ProcessState` as a reference - i.e., `process: &ProcessState`. The motivation is that we may want to pass the same state into multiple handlers and there is no need to re-create (or clone) the state between them.

Reply:
`ProcessState` currently implements `Copy` since it's just a collection of 3 pointers (and expected to remain lightweight). I could make it only `Clone`, but I think it's fine as is?

Reply:
Ah - no, it is fine as is then.